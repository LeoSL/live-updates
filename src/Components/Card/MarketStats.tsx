import { ComponentType } from "react";
import { makeStyles } from "@material-ui/core";

type StyleProps = {
  marketPositive: string;
};

const useStyles = makeStyles({
  percentColor: {
    color: (props: StyleProps) => props.marketPositive,
  },
});

const pickColor = (percentChange: number): string => {
  if (percentChange === 0) return "gray";
  if (percentChange > 0) return "green";
  return "red";
};

export const MarketStats: ComponentType<{ percentChange: number }> = ({
  percentChange,
}) => {
  const classes = useStyles({ marketPositive: pickColor(percentChange) });

  return percentChange === 0 ? (
    <span className={classes.percentColor}>{percentChange.toFixed(2)}%</span>
  ) : (
    <span className={classes.percentColor}>
      {percentChange > 0
        ? `ðŸ“ˆ +${percentChange.toFixed(2)}%`
        : `ðŸ“‰ ${percentChange.toFixed(2)}%`}
    </span>
  );
};
