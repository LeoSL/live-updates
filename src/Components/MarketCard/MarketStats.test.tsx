import { render } from "@testing-library/react";
import { MarketStats } from "./MarketStats";

it("renders correctly", () => {
  const { container } = render(<MarketStats percentChange={0.5} />);
  expect(container).toMatchSnapshot();
});
