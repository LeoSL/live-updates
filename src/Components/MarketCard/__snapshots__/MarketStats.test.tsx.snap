// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span
    class="makeStyles-percentColor-1 makeStyles-percentColor-2"
  >
    ðŸ“ˆ +0.50%
  </span>
</div>
`;
